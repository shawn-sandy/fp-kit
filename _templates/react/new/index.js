/* eslint-disable camelcase */
module.exports = {
  prompt: ({ inquirer }) => {
    const questions = [
      {
        type: 'input',
        name: 'component_name',
        message: 'What is the component name?'
      },
      {
        type: 'input',
        name: 'dir',
        message: 'The name of the directory to store the component (optional)'
      }
    ]
    return inquirer
      .prompt(questions)
      .then(answers => {
        const { component_name, dir } = answers
        const path = dir ? `${dir}` : `${component_name.toLowerCase()}`
        const absPath = `components/react/${path}`
        return { ...answers, path, absPath }
      })
  }
}
