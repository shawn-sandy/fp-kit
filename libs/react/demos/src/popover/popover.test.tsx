/* eslint-disable react/react-in-jsx-scope */
import { describe, expect, test } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Popover from './popover'

describe('Popover', () => {
  it('should show popover on button click', async () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    const component = render(<Popover />)
    const button = screen.getByText('Click me')
    fireEvent.mouseOver(button)
    expect(screen.getByText('This is a popover.')).toBeDefined()
    expect(component).toMatchSnapshot()
  })

  it('should hide popover on outside click', async () => {
    const component = render(<Popover />)
    const button = screen.getByText('Click me')
    fireEvent.mouseOver(button)

    expect(screen.getByText('This is a popover.')).toBeDefined()

    const container = screen.getByTestId('popover')

    fireEvent.mouseOut(container)
    expect(screen.queryByText('This is a popover.')).toBeNull()
    expect(component).toMatchSnapshot()
  })
})
