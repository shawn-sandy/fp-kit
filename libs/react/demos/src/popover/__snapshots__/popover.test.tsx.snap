// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > should hide popover on outside click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="popover"
      >
        <button>
          Click me
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="popover"
    >
      <button>
        Click me
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Popover > should show popover on button click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="popover"
      >
        <button>
          Click me
        </button>
        <div
          style="display: block; position: absolute; background: rgb(0, 0, 0); border: 1px solid #ccc; padding: 10px; color: rgb(255, 255, 255); top: 1px; left: 0px; transition: opacity .5s ease-in-out; opacity: 1; transform: translateY(0px);"
        >
          This is a popover.
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="popover"
    >
      <button>
        Click me
      </button>
      <div
        style="display: block; position: absolute; background: rgb(0, 0, 0); border: 1px solid #ccc; padding: 10px; color: rgb(255, 255, 255); top: 1px; left: 0px; transition: opacity .5s ease-in-out; opacity: 1; transform: translateY(0px);"
      >
        This is a popover.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
